### PR TITLE
QA-235: pipeline: Switch to static checks by golangci-lint

### PR DESCRIPTION
**Pull request by lluiscampos**
And fix the code accordingly.

### CODE REVIEW

File: store/mongo/datastore_mongo.go [hunk 1]
```diff
@@ -413,17 +426,17 @@ func (db *DataStoreMongo) GetAllJobs(
 	sortField := bson.M{}
 	sortField["insert_time"] = -1
 	findOptions.SetSort(sortField)
-	cur, err := collection.Find(ctx, bson.M{}, findOptions)
+	cur, _ := collection.Find(ctx, bson.M{}, findOptions)
```
Comment (tranchitella) [inline store/mongo/datastore_mongo.go:429]: We should return the error, not swallowing it:
```suggestion
      cur, err := collection.Find(ctx, bson.M{}, findOptions)
      if err != nil {
              return []model.Job{}, 0, err
      }
```

File: store/mongo/datastore_mongo.go [hunk 2]
```diff
@@ -413,17 +426,17 @@ func (db *DataStoreMongo) GetAllJobs(
...
-	err = cur.All(ctx, &jobs)
+	err := cur.All(ctx, &jobs)
 	if err == mongo.ErrNoDocuments {
 		return []model.Job{}, 0, nil
 	} else if err != nil {
 		return []model.Job{}, 0, err
 	}
 
-	count, err := collection.CountDocuments(ctx, bson.M{})
+	count, _ := collection.CountDocuments(ctx, bson.M{})
```
Comment (tranchitella) [inline store/mongo/datastore_mongo.go:439]: We should return the error, not swallowing it:
```suggestion
      count, err := collection.CountDocuments(ctx, bson.M{})
      if err != nil {
              return []model.Job{}, 0, err
      }
```

File: store/mongo/migration_1_0_0.go [hunk 3]
```diff
@@ -43,6 +43,7 @@ func (m *migration1_0_0) Up(from migrate.Version) error {
 	idxQueue := collQueue.Indexes()
 	idxJobs := collJobs.Indexes()
 	indexOptions := mopts.Index()
+	//nolint:staticcheck
 	indexOptions.SetBackground(false)
```
Comment (tranchitella) [inline store/mongo/migration_1_0_0.go:47]: You can safely remove `SetBackground(false)`

**Review by tranchitella — APPROVED** [review]
A few comments/suggestions

**Comment by lluiscampos** [comment]
@mender-test-bot start pipeline :three:

**Comment by lluiscampos** [comment]
@mender-test-bot start pipeline :three: :repeat:

**Comment by lluiscampos** [comment]
@mender-test-bot start pipeline :three: :repeat: :repeat:

**Comment by lluiscampos** [comment]
@tranchitella It seems like the rework on `worker.go` introduced some instability, some backend-tests were failing (see [here](https://gitlab.com/Northern.tech/Mender/mender-qa/-/jobs/1782123812), [here](https://gitlab.com/Northern.tech/Mender/mender-qa/-/jobs/1782688630) and [here](https://gitlab.com/Northern.tech/Mender/mender-qa/-/jobs/1787557594), not always the same test, which is even more worrying). So in my last force-push I reverted your patch (see [here](https://github.com/mendersoftware/workflows/compare/ca6c7eedfa404b8e778190b829c478a29694ccfe..42f933690bbbda162bbe8843188f35944e10b3b3)) and tests passed again.

I'll merge as-is but feel free to follow-up independently if you still want to clean-up further the code :+1:

**Pull request merged**